pkg/controllers/machinemigration/machine_migration_controller.go

```diff
@@ -346,76 +341,50 @@ func (r *MachineMigrationReconciler) requestOldAuthoritativeResourcePaused(ctx c
 	return updated, nil
 }
 
-func (r *MachineMigrationReconciler) isSynchronizedGenerationMatchingOldAuthoritativeAPIGeneration(ctx context.Context, mapiMachine *machinev1beta1.Machine) (bool, error) {
-	//nolint:wsl
+func (r *MachineMigrationReconciler) isSynchronized(ctx context.Context, mapiMachine *machinev1beta1.Machine) (bool, error) {
+	// Check if the Synchronized condition is set to True.
+	// If it is not, this indicates an unmigratable resource and therefore should take no action.
+	if cond, err := util.GetConditionStatus(mapiMachine, string(controllers.SynchronizedCondition)); err != nil {
+		return false, fmt.Errorf("unable to get synchronizedCondition for %s/%s: %w", mapiMachine.Namespace, mapiMachine.Name, err)
```

Comment:
I know this is my wrongdoing, but while you are at it, let's break down `synchronizedCondition` into two words
```suggestion
		return false, fmt.Errorf("unable to get synchronized condition for %s/%s: %w", mapiMachine.Namespace, mapiMachine.Name, err)
```